Fix Dependabot auto-merge workflow secret access

## Problem

The Dependabot auto-merge workflow is failing with the error:

```
gh: To use GitHub CLI in a GitHub Actions workflow, set the GH_TOKEN environment variable.
error: Process completed with exit code 4.
```

The workflow logs show that `GH_TOKEN` is empty even though `DEPENDABOT_PAT` is set in the workflow:

```yaml
env:
  GH_TOKEN: ${{ secrets.DEPENDABOT_PAT }}
```

## Root Cause

According to [GitHub's documentation](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions):

> When a workflow is initiated by Dependabot using the `pull_request` event, **only Dependabot secrets are available** - Actions secrets are not accessible.

The workflow currently uses the `pull_request` trigger, which means it cannot access Actions secrets when triggered by Dependabot PRs. The `DEPENDABOT_PAT` secret must exist as both:
1. **Actions secret** (for regular PRs) ✅ 
2. **Dependabot secret** (for Dependabot PRs) ✅ (just added)

## Solution

Now that `DEPENDABOT_PAT` has been added as a Dependabot secret, we need to trigger the workflow to run again on the Dependabot PR to verify it works.

## Testing Plan

1. Trigger the workflow on Dependabot PR #115 (pest update)
2. Verify the workflow can access `DEPENDABOT_PAT` and successfully:
   - Approve the PR
   - Enable auto-merge
3. Confirm the workflow completes without token errors

## Alternative Solution (Future Consideration)

Instead of requiring the secret in two places, we could change the workflow to use `pull_request_target` instead of `pull_request`. However, this has security implications:

- `pull_request_target` runs in the context of the base branch (main)
- Has access to all secrets
- But needs careful review to prevent security issues with untrusted code

For now, the Dependabot secrets approach is safer and more explicit.

## Acceptance Criteria

- [ ] Dependabot PR #115 workflow runs successfully
- [ ] Workflow can access `DEPENDABOT_PAT` secret
- [ ] PR is auto-approved and auto-merge is enabled
- [ ] No token-related errors in logs

## References

- [Dependabot PR #115](https://github.com/hemmer-io/jcl/pull/115)
- [GitHub Docs: Automating Dependabot with GitHub Actions](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions)
- [GitHub Docs: Dependabot secrets](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/configuring-access-to-private-registries-for-dependabot#storing-credentials-for-dependabot-to-use)

--- a/.github/workflows/dependabot-auto-merge.yml
+++ b/.github/workflows/dependabot-auto-merge.yml
@@ -89,7 +89,7 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
           wait-interval: 20
           running-workflow-name: 'Auto-Merge Dependabot PR'
-          allowed-conclusions: success
+          allowed-conclusions: success,skipped
 
       - name: Auto-approve PR
         if: steps.update-type.outputs.auto-merge == 'true'

